test/config_test.dart

```diff
@@ -180,6 +180,28 @@ void main() {
     }
   });
 
+  test("Invalid value for top-level property", () {
+    try {
+      var yamlString =
+          "name: foobar\n"
+          "port: 65536\n";
+
+      var _ = new TopLevelConfigurationWithValidation(yamlString);
```

Comment:
Heh, well you just made me realize I made a mistake on all of the tests - there should be a `fail` after this. If this somehow succeeded, the catch block wouldn't run and neither would the `expect` and the test would erroneously pass. I'm going to merge this and fix all of the tests since that was my bad.